pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -278,53 +204,69 @@ private QueryEnvironment.QueryPlannerResult getQueryPlannerResult(PlannerContext
     return new QueryPlannerResult(dispatchableSubPlan, explainStr, tableNames, extraFields);
   }
 
+  /// @deprecated Use [#compile] and then [explain][CompiledQuery#explain(long) ] the returned query instead
   @VisibleForTesting
+  @Deprecated
   public String explainQuery(String sqlQuery, long requestId) {
     SqlNodeAndOptions sqlNodeAndOptions = CalciteSqlParser.compileToSqlNodeAndOptions(sqlQuery);
-    QueryPlannerResult queryPlannerResult = explainQuery(sqlQuery, sqlNodeAndOptions, requestId, null);
-    return queryPlannerResult.getExplainPlan();
+    try (CompiledQuery compiledQuery = compile(sqlQuery, sqlNodeAndOptions)) {
+      QueryPlannerResult queryPlannerResult = compiledQuery.explain(requestId, null);
+      return queryPlannerResult.getExplainPlan();
+    }
   }
 
-  public List<String> getTableNamesForQuery(String sqlQuery) {
-    SqlNodeAndOptions sqlNodeAndOptions = CalciteSqlParser.compileToSqlNodeAndOptions(sqlQuery);
-    try (PlannerContext plannerContext = getPlannerContext(sqlNodeAndOptions)) {
+  public CompiledQuery compile(String sqlQuery) {
+    return compile(sqlQuery, CalciteSqlParser.compileToSqlNodeAndOptions(sqlQuery));
+  }
+
+  /// Given a query, parses, validates and optimizes the query into a [CompiledQuery].
+  ///
+  /// The returned query can then be planned, explained or used to get the tables involved in the query.
+  ///
+  /// @throws QueryException if the query cannot be compiled. Usual error types are [QueryErrorCode#SQL_PARSING] and
+  /// [QueryErrorCode#QUERY_VALIDATION]. [QueryErrorCode#QUERY_EXECUTION] is also possible if there is an error when
+  /// a function call is reduced into a constant.
```

Comment:
I don't know if it's just an issue with my IDE but these query error codes don't render in the Javadoc